lib/jormungandr/src/Cardano/Wallet/Jormungandr/Binary.hs

```diff
@@ -338,9 +345,43 @@ legacyUtxoWitness xpub bytes = TxWitness $ BL.toStrict $ runPut $ do
     putByteString (unXPub xpub)
     putByteString bytes
 
+-- | Decode the contents of a delegated transaction @Transaction@-message.
+getDelegatedCertificateTransaction
+    :: Hash "Tx"
+    -> Get (PoolId, ChimericAccount, Tx, [TxWitness])
+getDelegatedCertificateTransaction tid = do
+    (tx, wits) <- getGenericTransaction tid
+    accId <- getByteString 32
+    poolId <- getByteString 32
+    pure (PoolId poolId, ChimericAccount accId, tx, wits )
```

Comment:
According to the rust code, the typical structure for fragments is:

```
FRAGMENT = PAYLOAD{FRAGMENT-TYPE} | IOS | WITNESSES
```

Where the payload here is:

```
PAYLOAD{STAKE-DELEGATION} = ACCOUNT_PUBKEY (32 bytes) | PRAOS-VRF-PUBKEY (32 bytes)
```

So indeed, as @Anviking pointed out, I believe the certificate payload (a.k.a accId and poolId) should be deserialized before the inputs, outputs and witnesses.

(see also https://github.com/input-output-hk/chain-libs/blob/7a7c74359df7b7fbdb9f71a83c11f88100897c06/chain-impl-mockchain/doc/format.md#common-structure)